Mathlib/Tactic/LinearCombination.lean

```diff
@@ -191,17 +169,6 @@ Note: The left and right sides of all the equalities should have the same
 * `linear_combination (exp := n) e` will take the goal to the `n`th power before subtracting the
   combination `e`. In other words, if the goal is `t1 = t2`, `linear_combination (exp := n) e`
   will change the goal to `(t1 - t2)^n = 0` before proceeding as above.
-  This feature is not supported for `linear_combination2`.
-* `linear_combination2 e` is the same as `linear_combination e` but it produces two
-  subgoals instead of one: rather than proving that `(a - b) - (a' - b') = 0` where
-  `a' = b'` is the linear combination from `e` and `a = b` is the goal,
-  it instead attempts to prove `a = a'` and `b = b'`.
-  Because it does not use subtraction, this form is applicable also to semirings.
```

Comment:
Losing support for semirings feels like a bit of a shame; would it make sense to combine the syntaxes, but fallback to this approach internally if no ring can be found?

Reply:
I really think the tactic is *improved* by removing this from the scope.  I laid out my thoughts here
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Narrowing.20the.20scope.20of.20.60linear_combination.60/near/462845544
(and let's continue the discussion there rather than here), but for the record, let me write precisely what I think that scope should be:

`linear_combination` is a tactic which reduces problems of the form
```
h₁ : a₁ = b₁
h₂ : a₂ = b₂
h₃ : a₃ = b₃
⊢ A = B
```
to the task of showing that 
```
(f a₁ a₂ a₃ - f b₁ b₂ b₃) - (A - B)
```
equals zero (and then tries a standard normalization procedure to check that).  I believe the scope of the tactic should be limited to operations `f` for which this is the same thing as
```
f (a₁ - b₁) (a₂ - b₂) (a₃ - b₃) - (A - B)
```
Namely, **linear** operations `f`.

Reply:
To summarize where we landed after the Zulip discussion and some follow-up PRs:
- the PR #16103 added *native* support for cancellative semirings in `linear_combination`;
- the PR #16166 preserved (near-) backward compatibility (in particular the original syntax and the rudimentary support for non-cancellative semirings) by duplicating `linear_combination` to a tactic `linear_combination'` which will not be further modified.

Reply:
I don't think that removing support for `Nat` makes the tactic better. I would prefer the tactic to swap LHS and RHS as needed to avoid `-`.

Reply:
@urkud I'm not sure what you mean here -- as of #16103 Nat is supported natively (and `-` *means* swapping).  Can you say a bit more about your concern?

Reply:
I wrote my comment before the description of the scope change was added. Please ignore it.